specification.md

```diff
@@ -0,0 +1,131 @@
+# Feature Driven Architecture
+
+Feature Driven Architecture (FDA) is a structural framework that helps organize the codebase for visual interfaces in a way that lets you keep a constant development velocity while the amount of Features is growing.
+
+## Rationale
+
+We have various component models that let us compose the UI. We have multiple data flow models that simplify working with the data. And yet, we struggle to build an application that allows many people to work in parallel. We suffer an exponential slow-down when the codebase grows.
+
+The reason is the lack of agreement on splitting an application in a way that allows more people to work in parallel and keep the complexity under control.
+
+FDA is the missing piece that builds **on top** of existing component and data-flow models.
+
+Instead of organizing code by data and abstraction types, we want to organize the code based on the user-facing features that solve a user problem. This way, we believe it is easier to scale, communicate and navigate the codebase.
+
+## Goals
+
+**Discoverability and navigation** - see how the software is structured by looking at the file system. Help find any user-facing functionality in code.
+
+**Work parallelization** - enable working in parallel with multiple people. Reduce the number of merge conflicts. Reduce the number of discussions about where to put some piece of code.
+
+**Refactoring** - better understand the scope of a change. Gain much more confidence in changing a Feature by knowing it won't affect other Features. See what Features will be affected when changing a shared abstraction.
+
+**Experiments** - enable creating and removing experimental Features.what
+
+## Non-goals
+
+**Language and Framework** - it is language and framework agnostic. It leaves plenty of space to accommodate particular product needs and design decisions.
+
+**Data flow model** - it is data flow agnostic. Unidirectional, reactive, or actor data model doesn't matter as long as the principles are not broken.
+
+**Data fetching** - it is network layer agnostic.
+Websockets or HTTP, REST, RPC or GraphQL - doesn't matter as long as the principles are not broken.
+
+## How
+
+Feature Driven Architecture defines foundational boundaries: Feature, Cluster, Screen, Shared and Library. Each concept follows the core principles.
+
+A concept of a "Feature" is the foundation of Feature Driven Architecture. Everything else exists to support it.
+
+Breaking any principle described in this architecture will lead to specific problems in your codebase, and the goal here is to define the benefits and trade-offs.
```

Comment:
What are the trade-offs? Not required but I think it could be beneficial to highlight applications where this architecture really flourishes and applications where it might not (if any 😉)